src/cli.c

```diff
@@ -0,0 +1,57 @@
+#include "cli.h"
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <getopt.h>
+#include <stdio.h>
+
+#define MAX_HZ (1000000L)
+
+static const char* usage_str = "";
+
+__attribute__((__const__)) config_t* cli_config_default(void) {
```

Comment:
I think that this function does not qualify as const anymore, since it is allocating global memory (quote from GCC docu about const functions: "a function is not permitted to read global memory")
Perhaps it still is pure, although I doubt it. You could remove the attribute and see whether the compiler suggests the pure attribute.